presto-tests/src/main/java/com/facebook/presto/tests/datatype/DataType.java

```diff
@@ -135,6 +143,11 @@ private static String quote(String value)
         return "'" + value + "'";
     }
 
+    private static String binaryLiteral(SqlVarbinary value)
+    {
+        return "X'" + base16().encode(value.getBytes()) + "'";
```

Comment:
Traditionally databases have problems supporting standard for binary literals.. I'm keeping my finger crossed this won't fail us :)

Reply:
At the moment, this is only executed on Presto. The design of `DataType` and `DataTypeTest` doesn't seem to allow for database-specific literal conversion, so I'm not sure how else to do this. Any suggestions?

Reply:
`DataTypeTest` allows for swapping table setup role, i.e. creating with 'remote' database rather than in Presto.  However, in that case we need to speak with that 'remote' database DDL terminology (e.g. `bytea`) and not Presto's or the standard's. Thus, swapping table creation roles isn't necessarily single-line of change and can be accompanied with a functor producing correct literal in the 'remote' database... 

Did you try to write a test that writes `bytea` using Postgres JDBC and reading it from Presto?
(btw. this brings a though that supporting statement parameters instead of stringifying to sql literals would could be more robust for writing values.. at least where JDBC supports this)